### PR TITLE
Upgrade to angular 13.

### PR DESCRIPTION
**Pull request by mshima**
Angular 13 generated packages are esm now.
- Jest mock cannot mock an esm package, so switch to an alternative.
- Switch dayjs to use esm exports. Requires adjusts to tsconfig and jest.config.

Note: generated project is commonjs.
ESM support is done by tooling (typescript, angular compiler, jest, webpack)

Closes https://github.com/jhipster/generator-jhipster/issues/16960.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/angular/package.json [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
     "tslib": "2.3.1",
     "zone.js": "0.11.4"
   },
   "devDependencies": {
-    "@angular/cli": "12.2.12",
-    "@angular-builders/custom-webpack": "12.1.3",
-    "@angular-builders/jest": "12.1.2",
-    "@angular-eslint/eslint-plugin": "12.6.1",
+    "@angular/cli": "13.0.1",
+    "@angular-builders/custom-webpack": "13.0.0-beta.2",
```
Comment (qmonmert) [inline generators/client/templates/angular/package.json:21]: 13.0.0 available https://www.npmjs.com/package/@angular-builders/custom-webpack

**Comment by mshima** [comment]
`@angular-builders/*` version 13 was released but `@angular-builders/jest` shouldn't have.
It's not stable yet since it depends on a not stable jest-angular-preset:
https://github.com/just-jeb/angular-builders/blob/472a8c86ed9eb25735eca7f36f304332735f542c/packages/jest/package.json#L47

**Comment by DanielFran** [comment]
@mshima Is this ready and should be included in next release?

**Comment by mshima** [comment]
The PR is blocked on official ng-bootstrap support https://github.com/ng-bootstrap/ng-bootstrap/issues/4178 se we can drop `.npmrc` file again.
Tests passed due to a bug on mocha (probably). `npm test` was broken.

Hopefully we will have a release soon. Same as bootstrap 5.

**Comment by mshima** [comment]
Builds are failing due to https://github.com/ng-bootstrap/ng-bootstrap/pull/4185.

**Comment by DanielFran** [comment]
@mshima https://github.com/ng-bootstrap/ng-bootstrap/releases/tag/11.0.0 has been released

**Comment by mshima** [comment]
PR is ready.
Reviews are welcome.

File: generators/client/templates/angular/src/main/webapp/app/account/register/register.component.spec.ts.ejs [hunk 2]
```diff
@@ -67,10 +68,10 @@ describe('RegisterComponent', () => {
 
   it('should update success to true after creating an account', inject(
     [RegisterService<% if (enableTranslation) { %>, TranslateService<% } %>],
-    fakeAsync((service: RegisterService<% if (enableTranslation) { %>, mockLanguageService: TranslateService<% } %>) => {
+    fakeAsync((service: RegisterService<% if (enableTranslation) { %>, mockTranslateService: TranslateService<% } %>) => {
       jest.spyOn(service, 'save').mockReturnValue(of({}));
 <%_ if (enableTranslation) { _%>
-      mockLanguageService.currentLang = '<%= nativeLanguage %>';
+      mockTranslateService.currentLang = '<%= nativeLanguage %>';
```
Comment (Tcharl) [inline generators/client/templates/angular/src/main/webapp/app/account/register/register.component.spec.ts.ejs:74]: Not a mock anymore, right?

**Comment by Tcharl** [comment]
Everything LGTM: feel free to merge

**Review by DanielFran — APPROVED** [review]

**Pull request merged**